fastapi_log/models/api_log.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2025 Akretion (http://www.akretion.com).
+# @author Florian Mounier <florian.mounier@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from starlette.exceptions import HTTPException as StarletteHTTPException
+
+from odoo import api, fields, models
+
+
+class FastapiLog(models.Model):
+    _inherit = "api.log"
+
+    fastapi_endpoint_id = fields.Many2one(
+        comodel_name="fastapi.endpoint",
+        string="Endpoint",
+        ondelete="cascade",
+        index=True,
+    )
+
+    @api.model
+    def log_request(self, request, fastapi_environ=None, fastapi_endpoint=None):
+        log = super().log_request(request)
```

Comment:
2.
We leave the super method unchanged but we pass a copy of `request`

```suggestion
        log = super().log_request(request.copy())
```